src/Microsoft.Health.Fhir.Core/Features/Operations/IFhirOperationDataStore.cs

```diff
@@ -56,5 +56,13 @@ public interface IFhirOperationDataStore
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>A list of acquired export job.</returns>
         Task<IReadOnlyCollection<ExportJobOutcome>> AcquireExportJobsAsync(ushort maximumNumberOfConcurrentJobsAllowed, TimeSpan jobHeartbeatTimeoutThreshold, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Get compatible version.
+        /// </summary>
+        /// <param name="maxVersion">The maximum schema version specified by code</param>
+        /// /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The latest supported schema version from server.</returns>
+        Task<int> GetLatestCompatibleVersionAsync(int maxVersion, CancellationToken cancellationToken);
```

Comment:
In general things are looking good, some thoughts:
* Perhaps break this into its own interface `ISchemaMigrationDataStore`
* Think about how you can gracefully have the CosmosDb implementation return an unsupported status

Reply:
yes it was in my mind to separate this out. But I know Brandon is refactoring lot of files to break components to get sql out. So I am looking forward for his PR too. Anyways I will update it. Thank you!

Reply:
I think we should limit the schema stuff to the SQL packages. I'm not sure it has broad applicability to the Cosmos implementation and also the `_schema` routes only exist in the SQL packages.

Reply:
Addressed it.